### PR TITLE
PA-1322: check to stop execution if cluster is not in Healthy state

### PR DESCRIPTION
**Pull request by jkagliwal-px**
This change to handle the testcase execution in case - 
When the target cluster is not in "HEALTHY" state, skip running tests and exit. 

Jenkins Run - https://jenkins.pwx.dev.purestorage.com/job/PDS/job/pds-branch-build-test/445/

### CODE REVIEW

File: drivers/pds/targetcluster/targetcluster.go [hunk 1]
```diff
@@ -80,8 +82,12 @@ func (tc *TargetCluster) GetDeploymentTargetID(clusterID, tenantID string) (stri
 			deploymentTargetID = targetClusters[i].GetId()
 			log.Infof("deploymentTargetID %v", deploymentTargetID)
 			log.InfoD("Cluster ID: %v, Name: %v,Status: %v", targetClusters[i].GetClusterId(), targetClusters[i].GetName(), targetClusters[i].GetStatus())
+			targetClusterStatus = targetClusters[i].GetStatus()
 		}
 	}
+	if targetClusterStatus != "healthy" {
+		return "Target Cluster is not in healthy state, hence exiting the testcase execution", fmt.Errorf("error occured is: %v", err)
```
Comment (dbhatnagar-px) [inline drivers/pds/targetcluster/targetcluster.go:89]: @jkagliwal-px the return should be in the form of nil,<error> and not <string>,<error>. Also you can embed the error message in returned error itself.

**Review by dbhatnagar-px — CHANGES_REQUESTED** [review]

**Review by Madanagopal19 — APPROVED** [review]
LGTM

**Review by aravindranath14 — APPROVED** [review]
LGTM

**Review by dbhatnagar-px — APPROVED** [review]
Lgtm

**Pull request merged**